packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -898,17 +915,23 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         legendColor: this.state.color,
         shouldHighlight,
       });
-      const barHeight: number = Math.max(yBarScale(point.y), 0);
-      let adjustedBarHeight = 0;
-      if (barHeight <= 0) {
-        return <React.Fragment key={point.x instanceof Date ? point.x.getTime() : point.x}> </React.Fragment>;
-      } else if (barHeight <= Math.ceil(yBarScale(this._yMax) / 100.0)) {
-        adjustedBarHeight = Math.ceil(yBarScale(this._yMax) / 100.0);
-      } else {
-        adjustedBarHeight = barHeight;
+      const barHeight: number = this.props.supportNegativeYValues
+        ? yBarScale(point.y) - yBarScale(0)
+        : yBarScale(point.y);
+      // Calculate threshold for minimum visible bar height
+      const minBarHeight = Math.ceil(yBarScale(this._yMax - this._yMin) / 100.0);
+      let adjustedBarHeight = barHeight;
+
+      if (Math.abs(barHeight) === 0 || (barHeight < 0 && !this.props.supportNegativeYValues)) {
+        return <React.Fragment key={point.x as string}> </React.Fragment>;
+      }
+      // Adjust bar height if it's smaller than the threshold
+      else if (Math.abs(barHeight) <= minBarHeight) {
+        adjustedBarHeight = minBarHeight * Math.sign(barHeight);
```

Comment:
height cannot be negative. Keep it positive and a boolean to represent whether it is negative value or positive.

Reply:
I agree height cannot be negative but here it is being used as a variable. Using a boolean to keep track will make code a bit complex. Because height is being calculated at many places. We will have to put additional checks everywhere. So what do you suggest?